packages/web-runtime/src/pages/resolvePrivateLink.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+  <div class="oc-height-1-1 oc-link-resolve">
+    <h1 class="oc-invisible-sr">{{ pageTitle }}</h1>
+    <div class="oc-card oc-border oc-rounded oc-position-center oc-text-center oc-width-large">
+      <template v-if="loading">
+        <div class="oc-card-header">
+          <h2 key="private-link-loading">
+            <translate>Resolving private link…</translate>
+          </h2>
+        </div>
+        <div class="oc-card-body">
+          <oc-spinner :aria-hidden="true" />
+        </div>
+      </template>
+      <template v-else-if="errorMessage">
+        <div class="oc-card-header oc-link-resolve-error-title">
+          <h2 key="private-link-error">
+            <translate>An error occurred while resolving the private link</translate>
+          </h2>
+        </div>
+        <div class="oc-card-body oc-link-resolve-error-message">
+          <p class="oc-text-xlarge">{{ errorMessage }}</p>
+        </div>
+      </template>
+      <div class="oc-card-footer">
+        <p>
+          {{ configuration.currentTheme.general.slogan }}
+        </p>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  useRoute,
+  useRouteParam,
+  useStore,
+  useTranslations,
+  useRouter,
+  queryItemAsString,
+  useCapabilitySpacesEnabled
+} from 'web-pkg/src/composables'
+import { unref, defineComponent, computed, onMounted } from '@vue/composition-api'
+import { clientService } from 'web-pkg/src/services'
+import { createLocationSpaces } from 'files/src/router'
+import { dirname } from 'path'
+import { createFileRouteOptions } from 'web-pkg/src/helpers/router'
+import { useTask } from 'vue-concurrency'
+import {
+  buildShareSpaceResource,
+  buildSpace,
+  buildWebDavSpacesPath,
+  isMountPointSpaceResource,
+  isPersonalSpaceResource,
+  Resource,
+  SpaceResource
+} from 'web-client/src/helpers'
+import { DavProperty } from 'web-pkg/src/constants'
+import { urlJoin } from 'web-pkg/src/utils'
+import { configurationManager } from 'web-pkg/src/configuration'
+
+export default defineComponent({
+  name: 'ResolvePrivateLink',
+  setup() {
+    const store = useStore()
+    const router = useRouter()
+    const route = useRoute()
+    const id = useRouteParam('fileId')
+    const { $gettext } = useTranslations()
+    const hasSpaces = useCapabilitySpacesEnabled(store)
+
+    const pageTitle = computed(() => $gettext(unref(route).meta.title))
+    const configuration = computed(() => {
+      return store.getters.configuration
+    })
+
+    onMounted(() => {
+      resolvePrivateLinkTask.perform(queryItemAsString(unref(id)))
+    })
+
+    const resolvePrivateLinkTask = useTask(function* (signal, id) {
+      let path, resource
+      let matchingSpace = getMatchingSpace(id)
+      if (matchingSpace) {
+        path = yield clientService.owncloudSdk.files.getPathForFileId(id)
+        resource = yield clientService.webdav.getFileInfo(matchingSpace, { path })
+      } else {
```

Comment:
Since reviewing renamed files is not such a good UX: the `else` block (+ 2 additional functions below) is new. getting path and resource for files within own spaces has been moved to the if block above. The rest is untouched.